src/wallet/mod.rs

```diff
@@ -284,8 +286,8 @@ where
     }
 
     // Return derived address for the external descriptor at a specific index
```

Comment:
Just one nit, peek address can peek into both `external` and `internal` descriptors.
```suggestion
    // Return derived address for the descriptor of given keychain kind at a specific index
```